### PR TITLE
[codex] integrate DataGen web, synthetic, and curation paths

### PR DESCRIPTION
**Pull request by RaychevAngel**
## Summary
- Integrate Ron's Mode C web acquisition branch (#27), the synthetic SFT stack, and the curation/JSONL gate on top of the Tinker base branch (#35).
- Resolve the `src/data_generator/mode_c.py` vs `src/data_generator/mode_c/` file/package conflict by using the package layout.
- Keep #27's Tavily/search/crawl/extract path, add synthetic generation under `src/data_generator/mode_c/synthetic.py`, and expose both through `src.data_generator.mode_c`.
- Add explicit Mode C backend control with `DATA_GENERATOR_MODE_C_BACKEND=auto|synthetic|web`; `DATA_GENERATOR_SYNTHETIC_OFFLINE=1` forces synthetic without touching Tavily.
- Stop converting raw web pages into fake trainable assistant targets. Web acquisition handoffs are marked non-trainable until a real structuring stage produces chat/SFT labels.
- Route Manager's handoff conversion through the curation gate so Mode A, Mode B, and synthetic chat/SFT rows become trainable JSONL.
- Preserve Manager's programmatic `data_path` path without falling back to interactive `input()`.
- Add missing web acquisition dependencies: `tavily-python`, `trafilatura`, and `PyMuPDF`.
- Fix a syntax/indentation error in the existing #27 live Mode C web test.

## Why
#27, the earlier synthetic stack, and the earlier curation branch were individually useful but not merge-compatible as independent PRs. The big issue was ownership of Mode C: one branch made it a web-acquisition package, while another made it a synthetic-data module. This branch makes those roles explicit and compatible:

- #27: acquire raw web sources
- synthetic Mode C: synthesize trainable chat/SFT examples
- curation: normalize handoffs into Tinker-compatible JSONL

The follow-up hardening commit addresses the main integration risk found during overnight validation: raw web acquisition is useful source material, but it is not by itself supervised fine-tuning data. The curation gate now rejects targetless web rows instead of training on placeholder responses.

## Validation
- `python3 -m compileall src tests/test_mode_c.py tests/test_data_curation.py tests/test_manager.py`
- targeted DataGen/Manager suite -> `46 passed, 1 skipped`
- broad non-live suite excluding live HF/web/Tinker checks -> `169 passed, 4 skipped`
- live Tavily sanity: 1 basic search returned 1 result
- tiny live Mode C web acquisition: 1 Tavily query, 2 search results, 1 crawled page, source handoff written in memory
- tiny live Claude synthetic generation with strict validation: 8 records, teacher used, validation passed
- bounded live partial pipeline: Mode C synthetic/offline -> curation JSONL -> DecisionEngine `tinker_sft` plan -> 5-step live Tinker on `Qwen/Qwen3.5-9B`
- expanded stacked merge smoke through #48 -> `191 passed, 4 skipped`

## Not Run
- `tests/data_generator/test_mode_b_hf_retrieval.py`: live Hugging Face retrieval path is external/network-heavy and was intentionally excluded from broad local smoke commands.

## Stacking
Base: #35 (`codex/tinker-integration`). This PR supersedes the older draft splits #36, #37, and #38, which have now been closed with comments. Follow-up stacked PRs cover boundary tests, web structuring, Manager validation, and AutoResearch loop hardening.

### CODE REVIEW

**Comment by RaychevAngel** [comment]
Overnight live validation update:

- Ran the actual DataGen graph in strict Mode C web mode (`DATA_GENERATOR_MODE_C_BACKEND=web`, `DATA_GENERATOR_WEB_STRUCTURING=required`) using real Tavily search/crawl and teacher structuring.
- Result: 20 search results, 22 structured raw records, 22 curated chat/SFT rows.
- Dataset split from curation: 17 train / 2 val / 3 test.
- Follow-on live AutoResearch/Tinker smoke on this dataset completed: batch size 4 regressed, `learning_rate=0.0005` improved +59.9% and was kept.

Artifacts are local-only in the unpublished stack worktree under `outputs/overnight/live-datagen-graph-web/1779201994` and `outputs/overnight/live-autoresearch-webgraph-dataset/1779202168`. Conservative spend is tracked in #41.

**Comment by RaychevAngel** [comment]
Superseded by consolidated replacement PR(s): #115. This old draft was part of the fragmented DataGen/curation/Mode C stack.
